crates/router/src/db/payment_method.rs

```diff
@@ -105,44 +106,122 @@ impl PaymentMethodInterface for Store {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let payment_methods = self.payment_methods.lock().await;
+        let payment_method = payment_methods
+            .iter()
+            .find(|pm| pm.payment_method_id == payment_method_id)
+            .cloned();
+
+        match payment_method {
+            Some(pm) => Ok(pm),
+            None => Err(errors::StorageError::ValueNotFound(
+                "cannot find payment method".to_string(),
+            )
+            .into()),
+        }
     }
 
     async fn insert_payment_method(
         &self,
-        _m: storage::PaymentMethodNew,
+        m: storage::PaymentMethodNew,
     ) -> CustomResult<storage::PaymentMethod, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let mut payment_methods = self.payment_methods.lock().await;
+
+        let payment_method = storage::PaymentMethod {
+            #[allow(clippy::as_conversions)]
+            id: payment_methods.len() as i32,
+            customer_id: m.customer_id,
+            merchant_id: m.merchant_id,
+            payment_method_id: m.payment_method_id,
+            accepted_currency: m.accepted_currency,
+            scheme: m.scheme,
+            token: m.token,
+            cardholder_name: m.cardholder_name,
+            issuer_name: m.issuer_name,
+            issuer_country: m.issuer_country,
+            payer_country: m.payer_country,
+            is_stored: m.is_stored,
+            swift_code: m.swift_code,
+            direct_debit_token: m.direct_debit_token,
+            created_at: m.created_at,
+            last_modified: m.last_modified,
+            payment_method: m.payment_method,
+            payment_method_type: m.payment_method_type,
+            payment_method_issuer: m.payment_method_issuer,
+            payment_method_issuer_code: m.payment_method_issuer_code,
+            metadata: m.metadata,
+        };
+        payment_methods.push(payment_method.clone());
+        Ok(payment_method)
     }
 
     async fn find_payment_method_by_customer_id_merchant_id_list(
         &self,
-        _customer_id: &str,
-        _merchant_id: &str,
+        customer_id: &str,
+        merchant_id: &str,
     ) -> CustomResult<Vec<storage::PaymentMethod>, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let payment_methods = self.payment_methods.lock().await;
+        let payment_methods_found: Vec<storage::PaymentMethod> = payment_methods
+            .iter()
+            .filter(|pm| pm.customer_id == customer_id && pm.merchant_id == merchant_id)
+            .cloned()
+            .collect();
+
+        if payment_methods_found.is_empty() {
+            Err(
+                errors::StorageError::ValueNotFound("cannot find payment method".to_string())
```

Comment:
Same applies to here as well, 
```suggestion
Err(errors::ConnectorError:: PaymentMethodNotFound())
```